zkevm-circuits/src/evm_circuit/util/instrumentation.rs

```diff
@@ -0,0 +1,170 @@
+                    }
+                    CellType::Lookup(Table::Rw) => {
+                        report.rw_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Bytecode) => {
+                        report.bytecode_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Block) => {
+                        report.block_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Copy) => {
+                        report.copy_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Keccak) => {
+                        report.keccak_table = data_entry;
+                    }
+                    CellType::Lookup(Table::Exp) => {
+                        report.exp_table = data_entry;
+                    }
+                }
+            }
+            report_collection.push(report);
+        }
+        report_collection
+    }
+}
+
+/// Struct which contains a Cost/ColumnType report for a particular EVM
+/// `ExecutionStep`.
+#[derive(Clone, Debug, Default)]
+pub(crate) struct ExecStateReport {
+    pub(crate) state: ExecutionState,
+    pub(crate) storage_1: StateReportRow,
+    pub(crate) storage_2: StateReportRow,
+    pub(crate) storage_perm: StateReportRow,
+    pub(crate) byte_lookup: StateReportRow,
+    pub(crate) fixed_table: StateReportRow,
+    pub(crate) tx_table: StateReportRow,
+    pub(crate) rw_table: StateReportRow,
+    pub(crate) bytecode_table: StateReportRow,
+    pub(crate) block_table: StateReportRow,
+    pub(crate) copy_table: StateReportRow,
+    pub(crate) keccak_table: StateReportRow,
+    pub(crate) exp_table: StateReportRow,
+}
+
+impl From<ExecutionState> for ExecStateReport {
+    fn from(state: ExecutionState) -> Self {
+        ExecStateReport {
+            state,
+            ..Default::default()
+        }
+    }
+}
+
+impl From<&ExecutionState> for ExecStateReport {
+    fn from(state: &ExecutionState) -> Self {
+        ExecStateReport {
+            state: *state,
+            ..Default::default()
+        }
+    }
+}
+
+/// Struct that contains all of the measurament values required to evaluate the
+/// costs of a particular `ColumnType` of an `ExecStateReport`
+#[derive(Debug, Clone, Default)]
+pub(crate) struct StateReportRow {
+    pub(crate) avaliable_cells: usize,
+    pub(crate) unused_cells: usize,
+    pub(crate) used_cells: usize,
+    pub(crate) top_height: usize,
+    pub(crate) used_columns: usize,
+    pub(crate) utilization: f64,
+}
+
+impl From<(usize, usize, usize, usize, usize, f64)> for StateReportRow {
+    fn from(tuple: (usize, usize, usize, usize, usize, f64)) -> Self {
+        StateReportRow {
+            avaliable_cells: tuple.0,
+            unused_cells: tuple.1,
+            used_cells: tuple.2,
+            top_height: tuple.3,
+            used_columns: tuple.4,
+            utilization: tuple.5,
+        }
+    }
+}
```

Comment:
The usage of the From trait here looks complicated to me.  Why not have a constructor like `fn new(avaliable_cells: usize, unused_cells: usize, ...) -> Self`.  And then just use it like:
```rust
                let data_entry = StateReportRow::new(
                    total_avaliable_cells,
                    unused_cells,
                    *cells,
                    top_height,
                    cells / top_height,
                    _utilization,
                );
```

Or even better, remove this altogether and construct it like:
```rust
                let data_entry = StateReportRow {
                    tavaliable_cells: otal_avaliable_cells,
                    unused_cells,
                    used_cells: *cells,
                    top_height,
                    used_columns: cells / top_height,
                    utilization: _utilization,
                );
```

Reply:
Sure! I just did it to save LOC. But I agree is clearer on this way.